src/Collection.php

```diff
@@ -205,11 +205,11 @@ public function createDocument($document, $id = '', array $options = [])
      * Creates a new KuzzleDataMapping object, using its constructor.
      *
      * @param array $mapping Optional mapping
-     * @return DataMapping
+     * @return Mapping
      */
-    public function dataMappingFactory(array $mapping = [])
+    public function collectionMapping(array $mapping = [])
```

Comment:
Edit: if there is consensus over `collectionMapping` then I would rename all uses of the `Mapping` name (files, classes name, etc) into `CollectionMapping`, for the sake of consistency.